### PR TITLE
Fixed handling of empty values

### PR DESCRIPTION
**Pull request by Seb-C**

### CODE REVIEW

File: classes/driver/field/abstract.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ public function renderErrorValueHtml($value)
      * @param string $defaultValue
      * @return mixed
      */
-    public function getInputValue($defaultValue = '')
+    public function getInputValue($defaultValue = null)
```
Comment (ghost) [inline classes/driver/field/abstract.php]: Let me know if I am mistaken:

sanitazeValue casts 
(string) null
which gives an empty string, the previous $defaultValue ?

Comment (Seb-C) [reply classes/driver/field/abstract.php]: Yes, but the `sanitizeValue` method may be overriden by some fields that are not strings (like the input number).

**Review by jguyomard — APPROVED** [review]

File: classes/controller/admin/answer/appdesk.ctrl.php [hunk 2]
```diff
@@ -106,7 +106,7 @@ public function load_config()
 
                 // Gets header text
                 $headerText = \Arr::get($fieldAppdeskConfig, 'headerText', preg_replace('/\:\s*$/', ' ', $field->field_label));
-                $headerText = is_callable($headerText) ? $dataType($field) : $headerText;
+                $headerText = is_callable($dataType) ? $dataType($field) : $headerText;
```
Comment (shaoshiva) [inline classes/controller/admin/answer/appdesk.ctrl.php]: Are you sure of that ? I would have replaced `$dataType` with `$headerText` instead :
```
$headerText = is_callable($headerText) ? $headerText($field) : $headerText;
```

File: classes/driver/field/abstract.php (hunk 1 above)
Comment (shaoshiva) [reply classes/driver/field/abstract.php]: I agree with @Seb-C, the abstract method should be as neutral as possible, `null` as default value is better.

Could you juste update the PHPDoc according to your changes ? Maybe something like this :

```
@param mixed|null $defaultValue
```

File: classes/controller/admin/answer/appdesk.ctrl.php (hunk 2 above)
Comment (Seb-C) [reply classes/controller/admin/answer/appdesk.ctrl.php]: You're right, it seems more logical.

**Comment by Seb-C** [comment]
I added some minor fixes.
There is now a new bug though. Since the field label accepts a callback or function name, it's dangerous if some field is named like a real existing function (`date` for example).

**Comment by shaoshiva** [comment]
@Seb-C To prevent this kind of bug we may always use the prefixed field name, like `post_date` instead of `date`.

**Review by shaoshiva — APPROVED** [review]

**Pull request merged**